lib/programmemory.h

```diff
@@ -96,12 +96,12 @@ struct ExprIdToken {
     };
 };
 
-struct ProgramMemory {
+struct CPPCHECKLIB ProgramMemory {
     using Map = std::unordered_map<ExprIdToken, ValueFlow::Value, ExprIdToken::Hash>;
 
-    ProgramMemory() = default;
+    ProgramMemory() : mValues(new Map()) {}
 
-    explicit ProgramMemory(Map values) : mValues(std::move(values)) {}
+    explicit ProgramMemory(std::shared_ptr<Map> values) : mValues(std::move(values)) {}
```

Comment:
This should not need to be changed to a `shared_ptr`, you can implement it as:

```cpp
explicit ProgramMemory(Map values) : mValues(new Map()) {
    *mValues = std::move(values);
}
```

Reply:
Done. Did it even simpler by moving it into the constructor.